src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutItem.cs

```diff
@@ -73,5 +139,708 @@ internal void InvokeClick()
+				}
+				else
+				{
+					VisualStateManager.GoToState(this, "Focused", bUseTransitions);
+				}
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "Unfocused", bUseTransitions);
+			}
+
+			// CheckPlaceholderStates
+			if (hasToggleMenuItem && hasIconMenuItem)
+			{
+				VisualStateManager.GoToState(this, "CheckAndIconPlaceholder", bUseTransitions);
+			}
+			else if (hasToggleMenuItem)
+			{
+				VisualStateManager.GoToState(this, "CheckPlaceholder", bUseTransitions);
+			}
+			else if (hasIconMenuItem)
+			{
+				VisualStateManager.GoToState(this, "IconPlaceholder", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "NoPlaceholder", bUseTransitions);
+			}
+
+			// PaddingSizeStates
+			if (shouldBeNarrow)
+			{
+				VisualStateManager.GoToState(this, "NarrowPadding", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "DefaultPadding", bUseTransitions);
+			}
+
+			// We'll make the accelerator text visible if any item has accelerator text,
+			// as this causes the margin to be applied which reserves space, ensuring that accelerator text
+			// in one item won't be at the same horizontal position as label text in another item.
+			if (hasMenuItemWithKeyboardAcceleratorText && isKeyboardPresent)
+			{
+				VisualStateManager.GoToState(this, "KeyboardAcceleratorTextVisible", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "KeyboardAcceleratorTextCollapsed", bUseTransitions);
+			}
+		}
+
+		// Clear flags relating to the visual state.  Called when IsEnabled is set to false
+		// or when Visibility is set to Hidden or Collapsed.
+		 void ClearStateFlags()
+		{
+			m_bIsPressed = false;
+			m_bIsPointerLeftButtonDown = false;
+			m_bIsPointerOver = false;
+			m_bIsSpaceOrEnterKeyDown = false;
+			m_bIsNavigationAcceptOrGamepadAKeyDown = false;
+
+			UpdateVisualState();
+		}
+
+		// Create MenuFlyoutItemAutomationPeer to represent the 
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			return new MenuFlyoutItemAutomationPeer(this);
+		}
+
+		string GetPlainText()
+		{
+			return Text;
+		}
+
+		internal string KeyboardAcceleratorTextOverrideImpl
+		{
+			get
+			{
+				var pValue = KeyboardAcceleratorTextOverride;
+
+				// If we have no keyboard accelerator text already provided by the app,
+				// then we'll see if we can ruct it ourselves based on keyboard accelerators
+				// set on this item.  For example, if a keyboard accelerator with key "S" and modifier "Control"
+				// is set, then we'll convert that into the keyboard accelerator text "Ctrl+S".
+				if (pValue == null)
+				{
+					// UNO TODO
+					// KeyboardAccelerator.GetStringRepresentationForUIElement(this, pValue);
+
+					// If we were able to get a string representation from keyboard accelerators,
+					// then we should now set that as the value of KeyboardAcceleratorText.
+					if (pValue != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)